lib/inspec/errors.rb

```diff
@@ -11,4 +11,13 @@ class UnsatisfiedVersionSpecification < Error; end
   class DuplicateDep < Error; end
   class FetcherFailure < Error; end
   class ReporterError < Error; end
+
+  class AttributeError < Error; end
```

Comment:
The namespaces here are bit ugly - you might consider building a tree under Inspec::Attribute::Error, and then Inspec::Attribute::RequiredError, or similar.  Exceptions should usually end with the word Error, and be grouped together in a namespace tree.  AttributeRegistry might get its own branch.

Reply:
Also, consider adding attr_accessors to some of these - most of them would do well to have a slot for the attribute name involved, or the value provided, or the profile name.  Or, YAGNI, and string matching against the error message is good enough.